archaius2-core/src/test/java/com/netflix/archaius/ProxyFactoryTest.java

```diff
@@ -231,4 +237,63 @@ public void testWithLongMap() {
         sub2 = withArgs.getChildren().get("2");
         Assert.assertEquals(789, sub2);
     }
+    
+    public static interface ConfigWithCollections {
+        List<Integer> getList();
+        
+        Set<Integer> getSet();
+        
+        SortedSet<Integer> getSortedSet();
+        
+        LinkedList<Integer> getLinkedList();
+    }
+    
+    @Test
+    public void testCollections() {
+        SettableConfig config = new DefaultSettableConfig();
+        config.setProperty("list", "5,4,3,2,1");
+        config.setProperty("set", "5,4,3,2,1");
+        config.setProperty("sortedSet", "5,4,3,2,1");
+        config.setProperty("linkedList", "5,4,3,2,1");
+        
+        PropertyFactory factory = DefaultPropertyFactory.from(config);
+        ConfigProxyFactory proxy = new ConfigProxyFactory(config, config.getDecoder(), factory);
+        ConfigWithCollections withCollections = proxy.newProxy(ConfigWithCollections.class);
+        
+        Assert.assertEquals(Arrays.asList(5,4,3,2,1), new ArrayList<>(withCollections.getLinkedList()));
+        Assert.assertEquals(Arrays.asList(5,4,3,2,1), new ArrayList<>(withCollections.getList()));
+        Assert.assertEquals(Arrays.asList(5,4,3,2,1), new ArrayList<>(withCollections.getSet()));
+        Assert.assertEquals(Arrays.asList(1,2,3,4,5), new ArrayList<>(withCollections.getSortedSet()));
+        
+        config.setProperty("list", "6,7,8,9,10");
+        Assert.assertEquals(Arrays.asList(6,7,8,9,10), new ArrayList<>(withCollections.getList()));
+    }
+    
+    @Test
+    public void testCollectionsWithoutValue() {
+        SettableConfig config = new DefaultSettableConfig();
+        
+        PropertyFactory factory = DefaultPropertyFactory.from(config);
+        ConfigProxyFactory proxy = new ConfigProxyFactory(config, config.getDecoder(), factory);
+        ConfigWithCollections withCollections = proxy.newProxy(ConfigWithCollections.class);
+        
+        Assert.assertNull(withCollections.getLinkedList());
```

Comment:
afaik best practice for apis that return collections is to return an empty list / map / set rather than null; can we do that here?